### PR TITLE
Implementation of `CSX` and `DEUTSCH` gates

### PR DESCRIPTION
**Pull request by renatomello**
This PR is necessary for https://github.com/qiboteam/qibo/pull/963

### CODE REVIEW

**Review by AlejandroSopena — APPROVED** [review]

**Comment by renatomello** [comment]
Now this is also necessary to close https://github.com/qiboteam/qibo/issues/967

**Review by AlejandroSopena — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ classifiers=[
 [tool.poetry.dependencies]
 python=">=3.8.0,<3.12"
 numba=">=0.51.0"
-qibo=">=0.1.15"
+qibo={ git = "https://github.com/qiboteam/qibo.git" , branch = "new_gates" }
```
Comment (stavros11) [inline pyproject.toml:17]: This should be changed to what it was or remove the branch (use qibo master). If it doesn't work with the previous version we could consider making a release in qibo.

**Review by stavros11 — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (renatomello) [reply pyproject.toml]: It should work with `qibo` from `master` now, but not with the latest stable version

Comment (renatomello) [reply pyproject.toml]: Just to confirm: the current one is `0.1.15` and the next one is `0.1.16`. So it's `0.1.16`, right?

**Pull request merged**